cmd/libs/go2idl/make-client/generators/client-generator.go

```diff
@@ -0,0 +1,270 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package generators has the generators for the set-gen utility.
+package generators
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/args"
+	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
+	"k8s.io/kubernetes/cmd/libs/go2idl/namer"
+	"k8s.io/kubernetes/cmd/libs/go2idl/types"
+
+	"github.com/golang/glog"
+)
+
+// NameSystems returns the name system used by the generators in this package.
+func NameSystems() namer.NameSystems {
+	return namer.NameSystems{
+		"public":  namer.NewPublicNamer(0),
+		"private": namer.NewPrivateNamer(0),
+		"raw":     namer.NewRawNamer("", nil),
+	}
+}
+
+// DefaultNameSystem returns the default name system for ordering the types to be
+// processed by the generators in this package.
+func DefaultNameSystem() string {
+	return "public"
+}
+
+// Packages makes the sets package definition.
+func Packages(context *generator.Context, arguments *args.GeneratorArgs) generator.Packages {
+	boilerplate, err := arguments.LoadGoBoilerplate()
+	if err != nil {
+		glog.Fatalf("Failed loading boilerplate: %v", err)
+	}
+
+	// @lavalamp: This is quite hacky. Any suggestions?
+	topLevelTypes := map[*types.Type]struct{}{}
+	ref := context.Universe.Get(types.Name{"k8s.io/kubernetes/cmd/libs/go2idl/make-client/input", "ReferenceTypes"})
+	for _, m := range ref.Members {
+		topLevelTypes[m.Type] = struct{}{}
+	}
+	return generator.Packages{&generator.DefaultPackage{
```

Comment:
Regarding my comment above - what you would like to do, is exactly the same what in set-gen example is done.
So you
- generate "doc.go" file:
https://github.com/kubernetes/kubernetes/blob/master/cmd/libs/go2idl/set-gen/generators/sets.go#L71
- for every type (that is not filtered out), you generate a separate file, creating a dedicated generator for it:
https://github.com/kubernetes/kubernetes/blob/master/cmd/libs/go2idl/set-gen/generators/sets.go#L83